component/otelcol/connector/connector.go

```diff
@@ -0,0 +1,206 @@
+// Package connector exposes utilities to create a Flow component from
+// OpenTelemetry Collector connectors.
+package connector
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/internal/fanoutconsumer"
+	"github.com/grafana/agent/component/otelcol/internal/lazycollector"
+	"github.com/grafana/agent/component/otelcol/internal/lazyconsumer"
+	"github.com/grafana/agent/component/otelcol/internal/scheduler"
+	"github.com/grafana/agent/pkg/build"
+	"github.com/grafana/agent/pkg/util/zapadapter"
+	"github.com/prometheus/client_golang/prometheus"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelconnector "go.opentelemetry.io/collector/connector"
+	otelextension "go.opentelemetry.io/collector/extension"
+	sdkprometheus "go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/sdk/metric"
+
+	_ "github.com/grafana/agent/component/otelcol/internal/featuregate" // Enable needed feature gates
+)
+
+const (
+	ConnectorTracesToTraces = iota
+	ConnectorTracesToMetrics
```

Comment:
If we have them all eventually that's probably fine, no strong objections. 
Since these enums are not persistent anywhere, the values can probably change without anything breaking ;)